linux/ibus-keyman/tests/testfixture.cpp

```diff
@@ -220,12 +240,16 @@ get_involved_keys(IBusKeymanTestsFixture *fixture, km::tests::key_event test_eve
   // process key
   guint keyval = (test_event.modifier_state & KM_KBP_MODIFIER_SHIFT) ? gdk_keyval_to_upper(test_event.vk)
                                                                      : gdk_keyval_to_lower(test_event.vk);
+
+  // REVIEW: the keyval we use here are not correct for < 0x20 and >= 0x7f
+  // See /usr/include/X11/keysymdef.h and /usr/include/ibus-1.0/ibuskeysyms.h
```

Comment:
Can you explain which circumstances this is a problem?

Reply:
That's not a problem, at least not at the moment, because we don't use keyval. But if that ever changes then it becomes a problem because our tests will behave different than the real client.